database/merkledb/unit.go

```diff
@@ -0,0 +1,140 @@
+package merkledb
+
+const UnitSize = 16 + 1
```

Comment:
Can you add a spec for `UnitSize`? I still have a hard time reasoning about it  myself

Reply:
Added this comment to the description - I think this clears it up.

```
// UnitSize reflects the 16 possible values in a hex
// as a consequence each BranchNode will have 16 positions to store a Node.
// Given that a BranchNode can have the same Address of a LeafNode
// we add 1 position for that single exception.
// In position 17 or BranchNode[16] when theres a Node, it's a LeafNode.
```